zebra-network/src/peer/client.rs

```diff
@@ -221,13 +208,7 @@ impl Service<Request> for Client {
         match self.server_tx.try_send(ClientRequest { request, span, tx }) {
             Err(e) => {
                 if e.is_disconnected() {
-                    let ClientRequest { tx, .. } = e.into_inner();
-                    let _ = tx.send(Err(PeerError::ConnectionClosed.into()));
-                    future::ready(Err(self
-                        .error_slot
-                        .try_get_error()
-                        .expect("failed servers must set their error slot")))
-                    .boxed()
```

Comment:
What happens to the `ClientRequest.tx` we used to send on here?
What happens to the specific error in the `error_slot`?

Reply:
It's never sent to the `Connection` in the background so we never poll the `rx` so there's no reason to send an error through the `tx` before dropping it. I think this originally got added when we were figuring out the `MustUseSender`, and this was added to avoid panicking because the sender hadn't been used, but we don't even convert to a `MustUseSender` yet here so it should be fine.

There error slot here is gone by the same logic above, the error from a previous request doesn't need to be propagated back to subsequent failed attempts at new requests.